packages/core/src/config/configNormalizers/stackTrace.js

```diff
@@ -0,0 +1,161 @@
+/*
+ * (c) Copyright IBM Corp. 2025
+ */
+
+'use strict';
+
+const {
+  setLogger,
+  validateStackTraceMode,
+  validateStackTraceLength
+} = require('../configValidators/stackTraceValidation');
+
+const { DEFAULT_STACK_TRACE_MODE, DEFAULT_STACK_TRACE_LENGTH } = require('../../util/constants');
+
+/**
+ * @param {import('../../config').InstanaConfig} config
+ */
+exports.init = function init(config) {
+  setLogger(config.logger);
+};
+
+// Precedence order:
+//   agent > in-code > environment variable > default
+
+/**
+ *
+ * @param {import('../../config').InstanaConfig} config
+ * @returns {{ stackTrace: string, stackTraceLength: number }}
+ */
+exports.normalize = function normalize(config) {
+  if (!config) {
+    config = {};
+  }
+
+  if (!config.tracing) {
+    config.tracing = {};
+  }
+
+  const stackTrace = normalizeStackTraceMode(config.tracing);
+  const stackTraceLength = normalizeStackTraceLength(config.tracing);
+
+  return {
+    stackTrace,
+    stackTraceLength
+  };
+};
+
+/**
+ * Normalizes stack trace configuration from agent.
+ *
+ * @param {*} config - tracing.global config from agent containing stack-trace and stack-trace-length
+ * @returns {{ stackTrace: string | null, stackTraceLength: number | null }} - Normalized values
+ */
+exports.normalizeAgentConfig = function normalizeAgentConfig(config) {
+  if (!config || typeof config !== 'object') {
+    return { stackTrace: null, stackTraceLength: null };
+  }
+
+  const tracingObj = {
+    stackTrace: config['stack-trace'],
+    stackTraceLength: config['stack-trace-length']
+  };
+
+  const result = {
+    stackTrace: normalizeStackTraceMode(tracingObj, true),
+    stackTraceLength: normalizeStackTraceLength(tracingObj, true)
+  };
+
+  return result;
+};
+
+/**
+ * Normalizes stack trace mode configuration.
+ *
+ * @param {*} tracingObj - The tracing object containing stackTrace
+ * @param {boolean} [isAgentConfig=false] - Whether this is from agent config
+ * @returns {string | null} - Normalized value or null if from agent and invalid
+ */
+function normalizeStackTraceMode(tracingObj, isAgentConfig = false) {
+  const envVar = 'INSTANA_STACK_TRACE';
+
+  if (isAgentConfig) {
+    const configPath = 'stack-trace value from agent';
+
+    const validatedAgentConfig = validateStackTraceMode(tracingObj?.stackTrace, configPath);
+
+    if (validatedAgentConfig != null) {
+      return validatedAgentConfig;
+    }
+
+    return null;
+  }
+
+  const validatedConfigGlobal = validateStackTraceMode(
+    tracingObj?.global?.stackTrace,
```

Comment:
Its a bit confusing that for agent we forward the config value without global (see unannounced) but for incode we use the global usage here. That needs to be corrected, hard to follow the code.